source/common/request_stream_grpc_client_impl.cc

```diff
@@ -67,9 +67,10 @@ void RequestStreamGrpcClientImpl::onRemoteClose(Envoy::Grpc::Status::GrpcStatus
 }
 
 RequestPtr ProtoRequestHelper::messageToRequest(
-    const Envoy::Http::HeaderMap& base_header,
+    const Envoy::Http::RequestHeaderMap& base_header,
     const nighthawk::request_source::RequestStreamResponse& message) {
-  auto header = std::make_shared<Envoy::Http::HeaderMapImpl>(base_header);
+  auto header = std::make_shared<Envoy::Http::RequestHeaderMapImpl>();
+  header->copyFrom(*header, base_header);
```

Comment:
do you think this instantiation is something we're going to do in the future too?  Would it make sense to create a convenience method for this?  It happens twice in this PR, but if we expected it to happen in the future, too, it might make things easier for us.

Reply:
I suspect the inconvience was introduced on purpose in Envoy: drop this constructor to discourage copying. But that is a wild guess.. maybe we can bring up re-introducing that / ask why it was dropped? Otherwise we could consider proposing explicit ::clone() method(s)?